packages/teams-js/src/private/files.ts

```diff
@@ -296,6 +297,105 @@ export namespace files {
     extension: string;
   }
 
+  /**
+   * @hidden
+   * Hide from docs
+   *
+   * Actions specific to 3P cloud storage provider file and / or account
+   */
+  export enum CloudStorageProviderFileAction {
+    New = 'NEW',
```

Comment:
I don't think you actually need any of these other than Download/Upload/Delete. Except for Logout, the other ones are extra parameters inside of interfaces that could still be typesafe without the extra hint. For Logout, I think if you removed action from BaseCloudStorageProviderContent and renamed that to just "CloudStorageProviderContent" you would also be able to just pass that type to the logout function.

If you do this, then your CloudStorageProviderFileAction enum will only have enum types in it that callers actually need to provide (Download/Upload/Delete for the function on 755) and you will still have a bunch of typesafe functions

Reply:
Suggestion incorporated. Modified enum to only have types, which caller needs to provide actually. Consequently modified the required types and function argyments.